website/docs/automation/gitops/flux/ci.md

```diff
@@ -0,0 +1,230 @@
+---
+title: 'Continuous Integration and GitOps'
+sidebar_position: 50
+---
+
+We have successfully bootstrapped Flux on EKS cluster and deployed the application. To demonstrate how to make changes in the source code an application, build a new container images and leverage GitOps to deploy a new image to a cluster we introduce Continuous Integration pipeline. We will leverage AWS Developer Tools and [DevOps principles](https://aws.amazon.com/devops/what-is-devops/) to build [multi-architecture container images](https://aws.amazon.com/blogs/containers/introducing-multi-architecture-container-images-for-amazon-ecr/) for Amazon ECR.
+
+We created Continuous Integration Pipeline during the prepare environment step and now we need to make it up and running.
+
+![CI](assets/ci-multi-arch.png)
+
+First, clone CodeCommit repository for the application sources:
+
+```bash
+$ git clone ssh://${GITOPS_IAM_SSH_KEY_ID}@git-codecommit.${AWS_REGION}.amazonaws.com/v1/repos/${EKS_CLUSTER_NAME}-retail-store-sample ~/environment/retail-store-sample-codecommit
+```
+
+Next, populate the CodeCommit repository with the sources from the public repository of the [Sample application](https://github.com/aws-containers/retail-store-sample-app):
+
+```bash
+$ git clone https://github.com/aws-containers/retail-store-sample-app ~/environment/retail-store-sample-app
+$ git -C ~/environment/retail-store-sample-codecommit checkout -b main
+$ cp -R ~/environment/retail-store-sample-app/src ~/environment/retail-store-sample-codecommit
+$ cp -R ~/environment/retail-store-sample-app/images ~/environment/retail-store-sample-codecommit
+```
+
+We use AWS CodeBuild and define `buildspec.yml` to build new `x86_64` and `arm64` images in parallel.
```

Comment:
Maybe add a note on why we do this, put a link on Graviton on it's advantage ?

Reply:
Also make a ref to the terraform code so user can better understand how a single buildspec.yml result in 2 amd and arm build in the pipeline

Reply:
We have a link to terraform in the very beginning of the lab and also the diagram. I agree that we need more info on Graviton, but may be in the separate lab when we deploy built images to arm64 nodes.

@niallthomson @csantanapr Should we simplify this lab and build only x64 images and than introduce multi-arch separately with deploy to arm64 nodes?

Reply:
Lets keep arm64 image and add short sentence with a link to a graviton as informational